src/settings.rs

```diff
@@ -2,18 +2,97 @@ use crate::LOG_DRAIN;
 
 use serde::{Deserialize, Serialize};
 use slog::info;
+use std::collections::HashSet;
 
+#[derive(Serialize, Deserialize, Default, Debug)]
+#[serde(default)]
+pub(crate) struct IDRange {
+    pub min: i64,
+    pub max: i64,
+}
+
+#[derive(Serialize, Deserialize, Default, Debug)]
+#[serde(default)]
+pub(crate) struct RuleStrategy {
+    pub rule: String,
+    pub ranges: Vec<IDRange>,
+}
+
+fn valid_ranges(ranges: &[IDRange], prefix_error_message: &str) -> Result<(), String> {
+    for range in ranges {
+        if range.min > range.max {
+            return Err(format!(
+                "{}: min value greater than max value",
+                prefix_error_message
+            ));
+        }
+    }
+    Ok(())
+}
+
+fn valid_user_rule_settings(rule: &str, ranges: &[IDRange]) -> Result<(), String> {
+    let allowed_rules_values: HashSet<String> = vec![
+        "RunAsAny".to_string(),
+        "MustRunAs".to_string(),
+        "MustRunAsNonRoot".to_string(),
+    ]
+    .into_iter()
+    .collect();
+    if !allowed_rules_values.contains(rule) {
+        return Err(String::from("Invalid RunAsUser rule."));
+    } else if rule == "MustRunAs" && ranges.is_empty() {
+        return Err(String::from("Missing user ID range"));
+    }
+    valid_ranges(ranges, "Invalid user ID range")?;
+    Ok(())
+}
```

Comment:
Building the HashSet on every call is a bit intensive, we could do something like (rusty pseudo-code):

```rust
match rule {
    "RunAsAny" | "MustRunAsNonRoot" => Ok(()),
    "MustRunAs" => {
      if ranges.is_empty() {
          Err(anyhow!("Missing user ID range")
      } else {
          valid_ranges(ranges, "Invalid user ID range")
      }
    },
    _ => { Err(anyhow!("Invalid RunAsUser rule.") }
}
```